### PR TITLE
Save In Progress Environment Variables - OpenHack Feedback

### PR DESCRIPTION
**Pull request by alfredoihernandez**
Improve Developer Experience By Saving Environment Variables As They Are Created

Syncing Upstream From OpenHack Repos:
https://github.com/retaildevcrews/aks-secure-baseline-hack-template
https://github.com/asb-spark/aks-secure-baseline-hack

closes https://github.com/retaildevcrews/aks-secure-baseline-hack-template/issues/240

### CODE REVIEW

File: saveenv.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+echo '#!/bin/bash' > aks_baseline.env
+echo '' >> aks_baseline.env
+
+IFS=$'\n'
+
+for var in $(env | grep -E '_AKS_BASELINE' | sort | sed "s/=/='/")
+do
+  echo "export ${var}'" >> aks_baseline.env
+done
```
Comment (ckittel) [inline saveenv.sh:14]:
```suggestion
env | sed -n "s/\(_AKS_BASELINE=\)\(.*\)/\1'\2'/p" | sort >> aks_baseline.env
```
as a simple one-liner

File: saveenv.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+# change to the proper directory
+cd $(dirname $0)
+
```
Comment (ckittel) [inline saveenv.sh:5]:
```suggestion
```

probably don't need, right?

File: saveenv.sh [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+echo '#!/bin/bash' > aks_baseline.env
+echo '' >> aks_baseline.env
+
+IFS=$'\n'
```
Comment (ckittel) [inline saveenv.sh:9]:
```suggestion
echo $'#!/bin/bash\n\n' > aks_baseline.env
```

File: saveenv.sh [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+for var in $(env | grep -E '_AKS_BASELINE' | sort | sed "s/=/='/")
+do
+  echo "export ${var}'" >> aks_baseline.env
+done
+
```
Comment (ckittel) [inline saveenv.sh:15]: Or if you wanted to do it even cleaner (IMO), you can combine them all simply as.

```output
#!/bin/bash

cat > aks_baseline.env << EOF
#!/bin/bash

$(env | sed -n "s/\(_AKS_BASELINE=\)\(.*\)/\1'\2'/p" | sort)
EOF

cat aks_baseline.env
```

**Review by ckittel — CHANGES_REQUESTED** [review]
Hey there @alfredoihernandez -- Here is my review.

Mostly just simplification of what you've got.  At the end of the day, only a couple variables need to be captured (only those that live past the page they are captured on).

File: 03-aad.md [hunk 5]
```diff
@@ -21,22 +21,22 @@ Following the steps below you will result in an Azure AD configuration that will
 1. Query and save your Azure subscription's tenant id.
 
    ```bash
-   TENANTID_AZURERBAC=$(az account show --query tenantId -o tsv)
+   export TENANTID_AZURERBAC_AKS_BASELINE=$(az account show --query tenantId -o tsv)
```
[the remark this answers is not in the document]
Comment (ckittel) [reply 03-aad.md:24]: Based on our conversation last night, IGNORE this feedback.

The only place we need to mess with `export ` is for variables that currently do NOT have `export ` but need them so they can be captured in the `./saveenv.sh` script.  All others can stay unchanged from their current presentation.

File: saveenv.sh (hunk 4 above)
Comment (alfredoihernandez) [reply saveenv.sh:12]: done

Comment (alfredoihernandez) [reply saveenv.sh:12]: copy and paste worked great

**Review by ckittel — COMMENTED** [review]
--

**Review by ckittel — APPROVED** [review]
Nice work @alfredoihernandez! Thank you so much for the contribution.

**Pull request merged**